server/src/main/java/org/elasticsearch/action/bulk/Retry.java

```diff
@@ -160,9 +160,8 @@ private boolean canRetry(BulkResponse bulkItemResponses) {
             }
             for (BulkItemResponse bulkItemResponse : bulkItemResponses) {
                 if (bulkItemResponse.isFailed()) {
-                    final Throwable cause = bulkItemResponse.getFailure().getCause();
-                    final Throwable rootCause = ExceptionsHelper.unwrapCause(cause);
-                    if (!rootCause.getClass().equals(retryOnThrowable)) {
+                    final RestStatus status = bulkItemResponse.getFailure().getStatus();
```

Comment:
just to double check: we don't need to unwrap here anymore because the status of the root cause is propagated to its ancestor?

Reply:
Here we get the `status` field of `BulkItemResponse`'s `Failure`, it is a seperate field than `Exception cause` in `Failure` class. So I find it always has the good value (`RestStatus.TOO_MANY_REQUESTS`) ? because the exception type was changed only through `toXContent/fromXContent` of `BulkItemResponse`, but in it the `status` was already parsed seperatly. So the status should always be good ? (except after toXContent/fromXContent, the `BulkItemResponse` was transfered again using `readFrom/writeTo`, which I don't think it's the case ?)

If it's this, I changed `bulkItemResponse.getFailure().getStatus();` to `bulkItemResponse.status();`, because it's the same.

Reply:
I think this is correct. I also don't follow why `readFrom/writeTo` Would cause issues, the exception type does change but the status stays the same right?